core/src/components/cat-tag/cat-tag.tsx

```diff
@@ -0,0 +1,294 @@
+
+  /**
+   * Whether the select is disabled.
+   */
+  @Prop() disabled = false;
+
+  /**
+   * The placeholder text to display within the select.
+   */
+  @Prop() placeholder?: string;
+
+  /**
+   * Optional hint text(s) to be displayed with the select.
+   */
+  @Prop() hint?: string | string[];
+
+  /**
+   * A unique identifier for the input.
+   */
+  @Prop() identifier?: string;
+
+  /**
+   * The label for the select.
+   */
+  @Prop() label = '';
+
+  /**
+   * The name of the form control. Submitted with the form as part of a name/value pair.
+   */
+  @Prop() name?: string;
+
+  /**
+   * Visually hide the label, but still show it to assistive technologies like screen readers.
+   */
+  @Prop() labelHidden = false;
+
+  /**
+   * A value is required or must be checked for the form to be submittable.
+   */
+  @Prop() required = false;
+
+  /**
+   * Attributes that will be added to the native HTML input element.
+   */
+  @Prop() nativeAttributes?: { [key: string]: string };
+
+  /**
+   * The value of the control.
+   */
+  @Prop({ mutable: true }) value?: string[];
+
+  /**
+   * The validation errors for this input. Will render a hint under the input
+   * with the translated error message(s) `error.${key}`. If an object is
+   * passed, the keys will be used as error keys and the values translation
+   * parameters.
+   * If the value is `true`, the input will be marked as invalid without any
+   * hints under the input.
+   */
+  @Prop() errors?: boolean | string[] | ErrorMap;
+
+  /**
+   * Fine-grained control over when the errors are shown. Can be `false` to
+   * never show errors, `true` to show errors on blur, or a number to show
+   * errors on change with the given delay in milliseconds.
+   */
+  @Prop() errorUpdate: boolean | number = 0;
+
+  /**
+   * List of characters that should create a new tag. This need to be comparable to `keydownEvent.key`.
+   * Pasted values will also be split by those chars.
+   * Defaults to `[' ']`.
+   */
+  @Prop() tagCreationChars: string[] = [' ']
+
+
+  /**
+   * Emitted when the value is changed.
+   */
+  @Event() catChange!: EventEmitter<string[]>;
+
+  /**
+   * Emitted when the input received focus.
+   */
+  @Event() catFocus!: EventEmitter<FocusEvent>;
+
+  /**
+   * Emitted when the input loses focus.
+   */
+  @Event() catBlur!: EventEmitter<FocusEvent>;
+
+  @Listen('keydown')
+  onKeyDown(event: KeyboardEvent): void {
+    if (['Enter', ...this.tagCreationChars].includes(event.key)) {
+      event.preventDefault();
+      if (this.input?.value.trim() && !this.value?.includes(this.input?.value.trim())) {
+        this.value = [...(this.value ?? []), this.input.value.trim()]
+        this.catChange.emit(this.value);
+      }
+      if(this.input) {
+        this.input.value = ''
+      }
+    } else if (['Backspace'].includes(event.key) && this.input?.selectionStart === 0 && (this.value?.length ?? 0) > 0) {
```

Comment:
isnt this the same:  `(this.value?.length ?? 0) > 0` and `this.value?.length`

Reply:
no, the latter will render a 0 into the dom...